enhancements/mcd-reboot-dictionary.md

```diff
@@ -0,0 +1,262 @@
+  a single reboot
+
+## Proposal
+
+The proposal calls for a whitelist that MCD uses to decide which action to take
+after updating a configuration file for the MCO.  To prevent the admin from
+tampering with the whitelist, it will be included as part of the MCD image as a
+static file in yaml format.
+
+The information could be baked into the the MCD itself, but assuming we treat it
+as data, the file will be a list of entries containing:
+
+- filename or glob
+- path
+- action to perform
+- drain required
+- action specific data
+
+Example (and not necessarily 100% accurate) entries:
+
+| File | Path | Action | Drain | Args |
+| ---- | ---- | ------ | ----- | ---- |
+| * | /var/home/core/.ssh/ | none | false ||
+| *.service | /etc/systemd/system/ | systemctl | false | _restart_ |
+| kubelet.conf | /etc/kubernetes/ | systemctl | false | _reload_ kubelet.service |
+| crio.conf    | /etc/crio/ | systemctl | true | _restart_ crio.service ||
+| imaginary.conf | /somewhere/ | binary | false | /bin/my-custom-tool -q |
+| grub2.conf | /etc/ | reboot | true ||
+
+After the MCD writes out a configuation change, it will consult the whitelist
+before deciding what action is required. If the filename is present, has a
+valid action and any associated data, the action will be performed.  Otherwise,
+a reboot will be performed as in prior 4.x versions.
+
+If a file contains any options that require a reboot to be applied, then any change 
+to it is treated as requiring a reboot.
+
+If a file matches more than one entry, the first one listed will be applied.
+
+If no service is supplied for a ``systemctl`` action, the matching filename (with
+directory components stripped) will be treated as the name of the unit file to be
+operated on.
+
+The system's highest priority is to apply the changes, if the specified action
+results in an error, then a reboot will be performed as in prior 4.x versions.
+
+Errors include 
+* the failure of any API calls (eg. https://github.com/coreos/go-systemd ) necessary 
+  to apply the change
+* external binaries/scripts (eg. `systemctl`) returning a value other than ``0``
+* actions taking "too" long (eg. we might want to put a limit on drain operations)
+
+In the case of multiple files being changed by a single update, if any file requires
+a reboot (either by configuration or an error), then the whitelist is ignored for all
+(remaining) files.  Otherwise, all files are handled independantly.
+
+
+
+### User Stories [optional]
+
+#### Story 1
+
+As an admin, I want to update the SSH keys without any followup action, to 
+reduce the number of unnecessary periods of degraded system and application 
+availability.
+
+#### Story 2
+
+As an admin, I want to update the CRI-O configuration by restarting it's systemd
+unit, to reduce the number of unnecessary periods of degraded system and 
+application availability.
+
+#### Story 3
+
+As an admin, I want to update the kubelet config file without reboot, to 
```

Comment:
What change is desired and why is a drain not required for that change?

Reply:
The reply to your comment about Story 2 also applies here.

I'm going to leave Story 2 about CRI-O and update it to match the example in the Proposal (add drain).  

I'll make story 3 about ``chrony`` to illustrate the requirements for services that do not necessitate Pod drain.

Happy to discuss whether avoiding reboots for scenarios that require drain is appropriate for this enhancement or if it should be deferred for a subsequent one due to risk,  complexity, or correctness concerns.